direct-auth/OktaDirectAuth/OktaDirectAuth/Onboarding/Objects/User.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  User.swift
+//  OktaDirectAuth
+//
+//  Created by Mike Nachbaur on 2021-04-19.
+//
+
+import Foundation
+import OktaIdx
+
+class User: Codable {
+    let token: IDXClient.Token
+    lazy var profileInfo: [String:Any]? = {
+        token.profile
+    }()
+    
+    lazy var authenticatedAt: Date? = {
+        guard let epoch = profileInfo?["auth_at"] as? TimeInterval else { return nil }
+        return Date(timeIntervalSince1970: epoch)
+    }()
+    
+    lazy var expiresAt: Date? = {
+        guard let epoch = profileInfo?["exp"] as? TimeInterval else { return nil }
+        return Date(timeIntervalSince1970: epoch)
+    }()
+    
+    lazy var authenticatorMethods: [String]? = {
+        profileInfo?["amr"] as? [String]
+    }()
+    
+    init(with token: IDXClient.Token) {
+        self.token = token
+    }
+    
+    subscript(key: ProfileKey) -> String? {
+        profileInfo?[key.rawValue] as? String
+    }
+    
+    enum ProfileKey: String, CaseIterable {
```

Comment:
Should you only display these predefined profile key values? Why not display every single profile attribute we receive in the token claim instead of looking for specific keys only?

Reply:
This is for the sake of the sample app; I was trying to keep the implementation light-weight so I wouldn't need to do any full-fledged JWT parsing. Here's what it looks like in practice.

Ultimately I should pull in the JWT parser library to pull out these values better, but that doesn't (yet) have support for Swift Package Manager.

![Simulator Screen Shot - iPhone 12 Pro Max - 2021-04-19 at 11 41 33](https://user-images.githubusercontent.com/74688448/115286691-32411800-a104-11eb-866d-6c2821e532e4.png)